### ISSUE TITLE
Do you know the reason?

### ISSUE THREAD

**Issue by linhanxiao** [ijelliti/Deeplearning.ai-Natural-Language-Processing-Specialization#61]
In course 3 week 2 assignment, 
data_generator = data_generator(batch_size, v_sentences, v_labels, vocab['<PAD>'], True)
but in course 3 week 3 assignment,
data_generator = trax.supervised.inputs.add_loss_weights(
    data_generator(batch_size, v_sentences, v_labels, vocab['<PAD>'], True),
    id_to_mask=vocab['<PAD>'])
Why  the data_generator didn't mask padding in the loss weights of data? @ijelliti

**Comment by ijelliti** [comment]
@linhanxiao In order to train it in mini-batch, you need to pad the batch, so each batch has the same size. However, when you calculate the predicted accuracy, the loss function should not count those artificial additional.
